Sources/TuistCache/ContentHashing/MirrorHasher.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+class MirrorHasher {
+    let contentHashing: ContentHashing
+    init(contentHashing: ContentHashing) {
+        self.contentHashing = contentHashing
+    }
+
+    func hash(of target: Target) throws -> String {
+        var stringsToHash = [String]()
+        try calculateHashes(target, stringsToHash: &stringsToHash)
+        return try contentHashing.hash(stringsToHash.sorted())
+    }
+
+    private func calculateHashes( _ value: Any, name: String? = nil, stringsToHash: inout [String]) throws {
+        let mirror = Mirror(reflecting: value)
+        switch (value, mirror.displayStyle) {
```

Comment:
InfoPlist should also have a custom handling, because the `generatedFile` case should only consider the data